### PR TITLE
improve performance of MSSummary and MSMetaData

### PR DESCRIPTION
**Pull request by dmehring**
1. create MSMetaData::getFieldCodes()
2. avoid unnecessary table scan in MSMetaData::_getSubScanKeys() if _subScanProperties data structure has previously been cached
3. avoid unnecessary table scan in MSMetaData::getUniqueFiedIDs() if _subScanProperties data structure has previously been cached 

Performance increase of 15-20% realized.

### CODE REVIEW

File: ms/MSOper/MSSummary.cc [hunk 1]
```diff
@@ -1048,41 +1023,52 @@ void MSSummary::listField (LogIO& os, Record& outrec,  Bool verbose, Bool fillRe
...
-            uInt fld=fieldId(i);
-            if (fld<msFC.phaseDir().nrow()) {
-                MDirection mRaDec=msFC.phaseDirMeas(fld);
+        vector<MDirection> phaseDirs = _msmd->getPhaseDirs();
+        vector<String> fieldNames = _msmd->getFieldNames();
+        vector<String> codes = _msmd->getFieldCodes();
+        std::set<Int>::const_iterator fiter = uniqueFields.begin();
+        std::set<Int>::const_iterator fend = uniqueFields.end();
+        vector<Int> sourceIDs = _msmd->getFieldTableSourceIDs();
+        for (; fiter!=fend; ++fiter) {
```
Comment (juliantaylor) [inline ms/MSOper/MSSummary.cc]: this changes iteration order from as stored to sorted by field name right?
I assume this is intended?

Comment (juliantaylor) [inline ms/MSOper/MSSummary.cc]: actually nevermind, fiter is the uniquefield int set not the names ... so its sorted the same way

**Comment by juliantaylor** [comment]
looks good, thanks

**Comment by dmehring** [comment]
Thanks!

**Pull request merged**